evaluator.py

```diff
@@ -0,0 +1,186 @@
+        with open(config_path, encoding='utf-8') as file:
+            c = yaml.safe_load(file)
+
+        self.config = c
+        self.results = []
+
+    def eval_log(self, log_path):
+        with open(log_path, encoding='utf8') as file:
+            self.log = yaml.safe_load(file)
+
+        try:
+            metadata = self.log["tft-metadata"]
+            test_case_id = metadata["test_case_id"]
+            test_type = metadata["test_type"]
+            ft_result = self.log["result"]
+        except KeyError as e:
+            logger.error(f"KeyError: {e}. Malformed log handed to eval_log()")
+            raise Exception(f"eval_log(): error parsing {log_path} for expected fields")
+
+        bitrate_threshold = self.get_threshold(test_case_id, test_type)
+        bitrate_gbps = self.calculate_gbps(ft_result, test_type)
+
+        result = Result(
+            test_id = test_case_id,
+            test_type = test_type,
+            success = self.is_passing(bitrate_threshold, bitrate_gbps),
+            bitrate_gbps = bitrate_gbps)
+
+        self.results.append(result)
+
+    def is_passing(self, threshold: int, bitrate_gbps: Bitrate) -> bool:
+        return bitrate_gbps.tx >= threshold and bitrate_gbps.rx >= threshold
+
+    def get_threshold(self, test_case_id: TestCaseType, test_type: TestType) -> int:
+        try:
+            return self.config[test_type][TestCaseType[test_case_id].value]["threshold"]
+        except KeyError as e:
+            logger.error(f"KeyError: {e}. Config does not contain valid config for test case {test_type.name} id {test_case_id}")
+            raise Exception(f"get_threshold(): Failed to parse evaluator config")
+
+    def calculate_gbps(self, result: dict, test_type: TestType) -> Bitrate:
+        if test_type == TestType.IPERF_TCP.name:
+            return self.calculate_gbps_iperf_tcp(result)
+        elif test_type == TestType.IPERF_UDP.name:
+            return self.calculate_gbps_iperf_udp(result)
+        elif test_type == TestType.HTTP.name:
+            return self.calculate_gbps_http(result)
+        else:
+            logger.error(f"Error calculating bitrate, Test of type {test_type} is not supported")
+            raise Exception(f"calculate_gbps(): Invalid test_type {test_type} provided")
+
+    def dump_to_json(self) -> dict:
+        passing = []
+        failing = []
+        for result in self.results:
+            output = {"test_case_id": result.test_id,
+                      "test_type": result.test_type,
+                      "Bitrate_Gbps_rx": result.bitrate_gbps.rx,
+                      "Bitrate_Gbps_tx": result.bitrate_gbps.tx}
+            if result.success == True:
+                passing.append(output)
+            else:
+                failing.append(output)
+        return {"passing": passing, "failing": failing}
+
+    def calculate_gbps_iperf_tcp(self, result: dict) -> Bitrate:
+        # If an error occured, bitrate = 0
+        if "error" in result:
```

Comment:
I don't think so, although open to suggestions. This is a pretty concise way to check iperf results in my opinion since the iperf result is stored as json and input as a dict.

So essentially this line is looking at something like

```
json = {
    "tft-metadata": {
        "test_case_id": "POD_TO_POD_SAME_NODE",
        "test_type": "IPERF_UDP",
        "server": {
            "name": "worker-225",
            "pod_type": "SRIOV",
            "is_tenant": True,
            "index": 0,
        },
        "client": {
            "name": "worker-225",
            "pod_type": "SRIOV",
            "is_tenant": True,
            "index": 0,
        },
    },
    "command": " exec -t sriov-pod-worker-225-client-5201 -- iperf3 -c 10.131.1.27 -p 5201 --json -t 10 -u -b 25G",
    "result": {
        "start": {
            "connected": [],
            "version": "iperf 3.7",
            "system_info": "Linux sriov-pod-worker-225-client-5201 5.14.0-284.32.1.el9_2.x86_64 #1 SMP PREEMPT_DYNAMIC Wed Sep 6 08:21:33 EDT 2023 x86_64",
        },
        "intervals": [],
        "end": {},
        "error": "error - the server is busy running a test. try again later",
    },
}
```

And checking if the field json['result']['error'] exists, which it should if an error occured during iperf